jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+                                "d1_1", "d1_2", "d1_3", "d1_4", "d1_5", "d1_6",
+                                "d2_1", "d2_2", "d2_3", "d2_4", "d2_5", "d2_6"]
+            count = 1
+            for var in struct_variables:
+                if (not (count >= len(arr))):
+                    pass
+                    setattr(triad_struct, var, 0xFFFF & ((np.uint8(arr[count+1]) << 8) | (np.uint8(arr[count]))))
+                else:
+                    setattr(triad_struct, var, 0)
+                count += 2
+        except Exception as e:
+            print("triad exception:", e)
+            pass
+
+        # parse the spectral UART msg
+        # add in relevant error handling
+        # set the struct variables
+
+    def txt_handler(self, msg, struct):
+        '''
+        Prints info messages revieved from nucleo to screen
+        '''
+        print(msg)
+
+    def carousel_handler(self, msg, carousel_struct):
+        # msg format: <"$CAROUSEL,position>
+        try:
+            arr = msg.split(",")
+            carousel_struct.position = np.int8(arr[1])
+        except Exception as e:
+            print(e)
+
+    def heater_state_handler(self, msg, struct):
+        # msg format: <"$HEATER,device,enabled">
+        try:
+            arr = msg.split(",")
+            # Send back the heater and the state
+            struct.device = int(arr[1])
+            struct.enabled = bool(int(arr[2]))
+        except Exception as e:
+            print(e)
+
+    def heater_shutoff_handler(self, msg, struct):
+        # msg format: <"$auto_shutoff,device,enabled">
+        try:
+            arr = msg.split(",")
+            enabled = False
+            if (arr[1] == "1"):
+                enabled = True
+            print(enabled)
+            struct.auto_shutdown_enabled = enabled
+        except Exception as e:
+            print(e)
+
+    def heater_transmit(self, channel, msg):
+        # Upon Receiving a heater on/off command, send a command for the appropriate mosfet
+        print("Heater cmd callback")
+        struct = Heater.decode(msg)
+        message = "$Mosfet,{device},{enable},1111111"
+
+        # Heater/Nichromes are 5-7 so(assuming 0 index) add 7 to get the appropriate mosfet
+        translated_device = struct.device + 7
+        message = message.format(device=translated_device,
+                                 enable=int(struct.enabled))
+
+        while(len(message) < 30):
+            # Add an extra 1 for padding
+            message += "1"
+        print(message)
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+        pass
+
+    def heater_auto_transmit(self, channel, msg):
+        # Send the nucleo a message telling if auto shutoff for heaters is off or on
+        struct = HeaterAutoShutdown.decode(msg)
+        message = "$AutoShutoff,{enable},1111"
```

Comment:
same here... can you just add the 1s in the while loop

Reply:
yes